codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -124,15 +127,15 @@ protected String referenceStructureShape(StructureShape shape) {
           SmithyNameResolver.shapeNamespace(resourceOrService).concat(".");
       }
       if (!this.isOptional) {
-        return "%1$s{%2$s}".formatted(
+        return "return %1$s{%2$s}".formatted(
             namespace.concat(
               context.symbolProvider().toSymbol(serviceShape).getName()
             ),
             dataSource
           );
       }
       return """
-      func () *%s {
+      return func () *%s {
```

Comment:
I think we discussed that we don't need these inline function calls anywhere. Let's see if we can get it done cheaply. Otherwise add a TODO.

Reply:
I think if I just remove the function signature and make shape visitor return `return <Implementation>` this would "just" work as right now. However, I also think there could be a hidden failure that I am not thinking about right now. So, the best thing to do rn is to add TODO. When we have CI working fully and MPL in a stable state, I think at we should make an attempt to remove this anonymous function and test it.